api/src/main/java/org/apache/gravitino/authorization/RoleChange.java

```diff
@@ -152,4 +165,88 @@ public String toString() {
       return "REMOVESECURABLEOBJECT " + securableObject;
     }
   }
+
+  /**
+   * A UpdateSecurableObject to update securable object's privilege from role. <br>
+   * The securable object's metadata entity must same as new securable object's metadata entity.
+   * <br>
+   * The securable object's privilege must be different as new securable object's privilege. <br>
+   */
+  final class UpdateSecurableObject implements RoleChange {
+    private final SecurableObject securableObject;
+    private final SecurableObject newSecurableObject;
+
+    private UpdateSecurableObject(
+        SecurableObject securableObject, SecurableObject newSecurableObject) {
+      if (!securableObject.fullName().equals(newSecurableObject.fullName())) {
+        throw new IllegalArgumentException(
+            "The securable object's metadata entity must be same as new securable object's metadata entity.");
+      }
+      if (securableObject.privileges().containsAll(newSecurableObject.privileges())) {
+        throw new IllegalArgumentException(
+            "The securable object's privilege must be different as new securable object's privilege.");
```

Comment:
According to your code, the error message should be 'The updated securable object's privileges are just a part of that of old one'